geometry.scad

```diff
@@ -1772,13 +1915,27 @@ function _split_polygon_at_z(poly, z) =
 //   polys = A list of 3D polygons to split.
 //   xs = A list of scalar X values to split at.
 function split_polygons_at_each_x(polys, xs, _i=0) =
+    assert( is_consistent(polys) && is_path(poly[0],dim=3) ,
+            "The input list should contains only 3D polygons." )
+    assert( is_finite(xs), "The split value list should contain only numbers." )  
     _i>=len(xs)? polys :
     split_polygons_at_each_x(
         [
             for (poly = polys)
             each _split_polygon_at_x(poly, xs[_i])
         ], xs, _i=_i+1
     );
+    
+//***
+// all the functions split_polygons_at_ may generate non simple polygons even from simple polygon inputs:
```

Comment:
Please file an issue for this.

Reply:
Done. The case I reported as a fail in `split_polygons_at_  ` functions was wrong because it inputs a polygon and not a list of polygons  as required.